### PR TITLE
Implement configurable external recovery

### PR DESCRIPTION
**Pull request by danielgranhao**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 //! Most functionality can be accessed by creating an instance of [`LightningNode`] and using its methods.
 
 #![allow(clippy::let_unit_value)]
+#![allow(clippy::empty_line_after_doc_comments)]
```
Comment (gcomte) [inline src/lib.rs:8]: Either we do this, or the [already merged change](https://github.com/getlipa/lipa-lightning-lib/pull/1284), but better not both

File: src/backup.rs [hunk 2]
```diff
@@ -46,15 +46,23 @@ impl BackupManager {
             .map_to_permanent_failure("Failed to perform backup of local db")
     }
 
-    pub async fn recover(&self) -> Result<()> {
+    pub async fn recover(&self, external_recovery_is_allowed: bool) -> Result<()> {
```
Comment (gcomte) [inline src/backup.rs:49]: Why not just `allow_external_recovery`?

File: src/lib.rs (hunk 1 above)
Comment (danielgranhao) [reply src/lib.rs:10]: Let's use my solution. Changing the CI toolchain version does not fix the issue on local environments. 

I've added a comment explaining why this is done.

File: src/backup.rs (hunk 2 above)
Comment (danielgranhao) [reply src/backup.rs]: Applied in [69bd723](https://github.com/getlipa/lipa-lightning-lib/pull/1283/commits/69bd72318f569863776ccb56791c5a9c009528aa)

**Review by gcomte — APPROVED** [review]

**Pull request merged**